indico/modules/events/registration/util.py

```diff
@@ -524,40 +567,73 @@ def update_regform_item_positions(regform):
             child.position = next(positions if child_active else disabled_positions)
 
 
+def map_header_to_fields(header, regform):
+    """Map CSV header item to field name for import
+
+    returns None to indicate the column is unknown and should be skipped
+    """
+    header = to_unicode(header).strip()
+    if header == 'Id':
+        return None
+    if ':' in header:
+        html_field_name, field_title = header.split(':')
+    else:
+        html_field_name = field_title = header
+    for item in regform.active_fields:
+        if html_field_name == item.html_field_name:
+            return html_field_name
+        if field_title == item.parent.title + '#' + item.title:
+            return item.html_field_name
+    else:
+        if html_field_name in ['reg_date','state', 'price', 'checked_in',
+                               'checked_in_date', 'payment_date']:
+            return html_field_name
+    return None
+
+
 def import_registrations_from_csv(regform, fileobj, skip_moderation=True, notify_users=False):
     """Import event registrants from a CSV file into a form."""
     reader = csv.reader(fileobj.read().splitlines())
     query = db.session.query(Registration.email).with_parent(regform).filter(Registration.is_active)
     registered_emails = {email for (email,) in query}
     used_emails = set()
     todo = []
-    for row_num, row in enumerate(reader, 1):
-        try:
-            first_name, last_name, affiliation, position, phone, email = [to_unicode(value).strip() for value in row]
-            email = email.lower()
-        except ValueError:
-            raise UserValueError(_('Row {}: malformed CSV data - please check that the number of columns is correct')
-                                 .format(row_num))
-
-        if not email:
-            raise UserValueError(_('Row {}: missing e-mail address').format(row_num))
-        if not validate_email(email):
-            raise UserValueError(_('Row {}: invalid e-mail address').format(row_num))
-        if not first_name or not last_name:
-            raise UserValueError(_('Row {}: missing first or last name').format(row_num))
-        if email in registered_emails:
-            raise UserValueError(_('Row {}: a registration with this email already exists').format(row_num))
-        if email in used_emails:
-            raise UserValueError(_('Row {}: email address is not unique').format(row_num))
-
-        used_emails.add(email)
-        todo.append({
-            'email': email,
-            'first_name': first_name.title(),
-            'last_name': last_name.title(),
-            'affiliation': affiliation,
-            'phone': phone,
-            'position': position
-        })
+    keys = []
+    for row_num, row in enumerate(reader):
```

Comment:
insert of the big if/else inside the loop better go for something like this:

```
header_row = next(reader)
# ... handle the header data
for row_num, row in enumerate(reader, 1):
    ... # logic to handle the registrant data
```

Reply:
good idea. I'll try to prepare something (it may take a bit as I am currently busy with other things),